src/lib/languages/mod.rs

```diff
@@ -0,0 +1,20 @@
+use super::*;
+mod javascript;
+pub use javascript::JavaScript;
+
+#[derive(Debug)]
+pub struct LangError {
+  language: String
+}
+
+pub fn generate(parser: Parser, lang: &str) -> Result<String, String> {
```

Comment:
Maybe we can better change lang into an enum.

Reply:
I thought I'd leave this for now, as I have not worked with enums much. Also, I am not sure how you wish to parse command line arguments. How will the end user choose the language to compile into? I'm not sure how I'd implement this, but if you have an idea, please let me know :-)

Reply:
Ah this would be a great time to create one then :)  
```rust
enum Lang {
  JS,
}
```

To convert a string into a lang you have a few options:  
1. Using Into (This is usually recommended when you are sure the into is possible so the parsing should not result in errors)  
```rust
impl Into<Lang> for String {
```

2. Using a generic function and match
```rust
impl Lang {
  fn parse(input: String) -> Result<Self, ParsingError> {
    match input {
       // ...
```

Reply:
I think I understand what you mean. My latest commit (https://github.com/talpalang/talpa/pull/4/commits/921de36219132e422eb74418e7228f3dfc52afa9) adds the Lang enum and an `Into` implementation.